docs/ROOTFS_DISCOVERY.md

```diff
@@ -0,0 +1,395 @@
+boot loader on the local disk, but use UKIs and `rootfs` from networked souces;
+or alternatively place both boot loader and UKIs on the local disk, and only
+the `rootfs` on a networked source.
+
+## Trust & Security
+
+In a modern boot integrity world, all three of the relevant components as well
+as (most of) their sidecars require cryptographic protection before they can be
+consumed. Specifically:
+
+1. The boot loader is typically authenticated by the firmware before invocation
+   via UEFI Secure Boot, i.e. checked against a cryptographic certificate list
+   persistently stored in the firmware. Note that the various auxiliary
+   resources available the `systemd-boot` boot loader reads are not
+   individually authenticated (i.e. `loader.conf` as well as Type #1 Boot
+   Loader Specification entries). Because of this they can typically only be
+   used in a very restricted fashion, i.e. configure some UI details as well as
+   menu entries. Some options available are ignored if Secure Boot mode is
+   enabled. Moreover, even if the text strings shown in the menu entries might
+   not be authenticated the binaries that are invoked once they are selected
+   are.
+
+2. The UKIs are also authenticated by the firmware via UEFI Secure Boot, and so
+   are EFI Addons. `confext` and `sysext` sidecars are protected via
+   `dm-verity` and a signature of the root hash is validated against keys in
+   the kernel's built-in keyring. System credentials are authenticated via
+   secrets stored in the TPM.
+
+3. Authentication of the `rootfs` and `usrfs` is more variable: depending on
+   setup this is either done via `dm-verity` (either pinned by root hash from
+   the UKI, or authenticated by signature prvoided to kernel, checked against
+   the kernel's built-in keyring) or `dm-crypt`+`dm-integrity` (protected by
+   TPM or user provided password/FIDO/PKCS#11), or in the network case at
+   download time via detached signatures (currently only GPG) or via HTTPS
+   certificate validation. Note that by default the automatic discovery
+   mechanism of the `rootfs` and its auxiliary file systems does not insist on
+   cryptographic protection and authentication before use. However, the
+   [`systemd.image_policy=`](https://www.freedesktop.org/software/systemd/man/latest/systemd.image-policy.html)
+   kernel command line switch may be used to control precisely what kind of
+   protection to require from each such partition.
+
+Note that UEFI Secure Boot is problematic in various ways: it is generally
+bound to a certificate list maintained centrally by Microsoft, and thus implies
+a complex (and expensive) code signing bureaucracy, that in many cases is
+undesirable, particularly in a community Linux world. Moreover, because he
+certficate list is very large its security value is limited: it mostly acts
+more as denylist of known bad software rather than as allowlist of known good.
```

Comment:
Please mention that is only relates to UEFI secure boot when using the microsoft keys. If you enroll your own keys you don't have these issues.